src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/updateotherparty/UpdateOtherPartyMidEventHandler.java

```diff
@@ -0,0 +1,102 @@
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Address;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.YesNo;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.ccd.validation.address.PostcodeValidator;
+
+@Service
+@Slf4j
+public class UpdateOtherPartyMidEventHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    private final PostcodeValidator postcodeValidator = new PostcodeValidator();
+
+    private static final String OTHER_PARTY = "other party";
+    private static final String OTHER_PARTY_REPRESENTATIVE = "other party representative";
+    private static final String ERROR_ADDRESS_LINE_1 = "You must enter address line 1 for the %s";
+    private static final String ERROR_COUNTRY = "You must enter a valid country for the %s";
+    private static final String ERROR_POSTCODE = "You must enter a valid UK postcode for the %s";
+    private static final String ERROR_MAINLAND_SELECTION = "You must select whether the address is in mainland UK for the %s";
+    private static final String ERROR_ADDRESS_MISSING = "Address details are missing for the %s";
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.MID_EVENT)
+                && callback.getEvent() == UPDATE_OTHER_PARTY_DATA
+                && isIbcaCase(callback.getCaseDetails().getCaseData())
+                && nonNull(callback.getCaseDetails().getCaseData().getOtherParties());
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+        Set<String> validationErrors = new HashSet<>();
+
+        caseData.getOtherParties().forEach(party -> {
+            validateAddress(party.getValue().getAddress(), OTHER_PARTY, validationErrors);
+
+            if (party.getValue().hasRepresentative()) {
+                validateAddress(
+                        party.getValue().getRep().getAddress(),
+                        OTHER_PARTY_REPRESENTATIVE,
+                        validationErrors
+                );
+            }
+        });
+
+        response.addErrors(validationErrors);
+        return response;
+    }
+
+    private void validateAddress(Address address, String addressPrefix, Set<String> validationErrors) {
```

Comment:
instead of passing validationErrors and mutating it within the method, why not return the error HashSet

Reply:
I initiated it here outside of validate so when I run it for otherParty and otherParty Rep, it'll use the single hashSet, rather than have to return two hashSets